app/controlplane/plugins/core/guac/v1/guac.go

```diff
@@ -0,0 +1,268 @@
+	i.Logger.Info("execution requested")
+
+	// Extract registration and attachment configuration if needed
+	var registrationConfig *registrationState
+	if err := sdk.FromConfig(req.RegistrationInfo.Configuration, &registrationConfig); err != nil {
+		return fmt.Errorf("invalid registration configuration %w", err)
+	}
+
+	client, err := storage.NewClient(ctx, option.WithCredentialsJSON([]byte(req.RegistrationInfo.Credentials.Password)))
+	if err != nil {
+		return fmt.Errorf("creating storage client: %w", err)
+	}
+
+	bucket, err := loadBucket(ctx, client, registrationConfig.Bucket)
+	if err != nil {
+		return fmt.Errorf("loading bucket: %w", err)
+	}
+
+	// 1 - Upload the attestation
+	envelopeJSON, err := json.Marshal(req.Input.Attestation.Envelope)
+	if err != nil {
+		return fmt.Errorf("marshalling attestation: %w", err)
+	}
+
+	// chainloop-deadbeef-attestation.json
+	fileName := fmt.Sprintf("%s-%s-attestation.json", filePrefix, req.Input.Attestation.Hash.Hex)
+	if err := uploadToBucket(ctx, bucket, fileName, envelopeJSON, req.ChainloopMetadata, i.Logger); err != nil {
+		return fmt.Errorf("uploading the SBOM to the bucket: %w", err)
+	}
+
+	// 2 - Upload all the materials, in our case they are SBOMs
+	for _, sbom := range req.Input.Materials {
+		// Validate that in fact they are SBOMs
+		if err := validateExecuteOpts(sbom, req.RegistrationInfo, req.AttachmentInfo); err != nil {
+			i.Logger.Warnw("msg", "invalid material received", "err", err)
+			continue
+		}
+
+		// chainloop-deadbeef-sbom.json
+		fileName := fmt.Sprintf("%s-%s-%s", filePrefix, sbom.Hash.Hex, sbom.Value)
+		if err := uploadToBucket(ctx, bucket, fileName, sbom.Content, req.ChainloopMetadata, i.Logger); err != nil {
+			return fmt.Errorf("uploading the SBOM to the bucket: %w", err)
+		}
+	}
+
+	i.Logger.Info("execution finished")
+	return nil
+}
+
+// uploadToBucket uploads the provided content to the bucket under a deterministic, unique name (digest + filename)
+// It also sets the content type and additional metadata
+func uploadToBucket(ctx context.Context, bucket *storage.BucketHandle, fileName string, content []byte, md *sdk.ChainloopMetadata, logger *log.Helper) error {
+	bucketInfo, err := bucket.Attrs(ctx)
+	if err != nil {
+		return fmt.Errorf("getting bucket info: %w", err)
+	}
+
+	buf := bytes.NewBuffer(content)
+	fileSize := uint64(buf.Len())
+	logger.Infow("msg", "writing to the bucket", "file", fileName, "bucket", bucketInfo.Name, "size", bytefmt.ByteSize(fileSize))
+
+	w := bucket.Object(fileName).NewWriter(ctx)
+	defer w.Close()
+
+	// Set metadata and content type
+	// application/json can't be detected automatically by https://mimesniff.spec.whatwg.org/#supplied-mime-type-detection-algorithm
+	// If not set, the underlying library will try to sniff it automatically
+	if filepath.Ext(fileName) == ".json" {
+		w.ObjectAttrs.ContentType = "application/json"
+	}
+
+	w.ObjectAttrs.Metadata = map[string]string{
+		"author":          "chainloop",
```

Comment:
chainloop.dev

Reply:
I though about it and decided not to add the domain on purpose but instead the project name. I could change it if you think it's much better.

Reply:
I think this is better and we make sure it is unique.